jina/enums.py

```diff
@@ -271,6 +271,14 @@ class ProviderType(BetterEnum):
     SAGEMAKER = 1  #: AWS SageMaker
 
 
+class ProviderEndpointType(BetterEnum):
+    """Provider endpoint type."""
+
+    NONE = 0
+    RANK = 1
+    ENCODE = 2
```

Comment:
Any reason we have rank & encode hard-coded here?

Reply:
oh yes, this should not be an enum. This should be free

Reply:
ok I can remove that

Reply:
pls remove yes